pydantic_factories/value_generators/regex.py

```diff
@@ -0,0 +1,135 @@
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of the Leapfrog Direct Response, LLC, including
+      its subsidiaries and affiliates nor the names of its
+      contributors, may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL LEAPFROG DIRECT
+RESPONSE, LLC, INCLUDING ITS SUBSIDIARIES AND AFFILIATES, BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""
+
+from itertools import chain
+from random import Random
+from sre_parse import SubPattern, parse  # pylint: disable=W4901,E0611
+from string import (
+    ascii_letters,
+    ascii_lowercase,
+    ascii_uppercase,
+    digits,
+    printable,
+    punctuation,
+    whitespace,
+)
+from typing import Any, Dict, List, Optional, Pattern, Tuple, Union
+
+_alphabets = {
+    "printable": printable,
+    "letters": ascii_letters,
+    "uppercase": ascii_uppercase,
+    "lowercase": ascii_lowercase,
+    "digits": digits,
+    "punctuation": punctuation,
+    "nondigits": ascii_letters + punctuation,
+    "nonletters": digits + punctuation,
+    "whitespace": whitespace,
+    "nonwhitespace": printable.strip(),
+    "normal": ascii_letters + digits + " ",
+    "word": ascii_letters + digits + "_",
+    "nonword": "".join(set(printable).difference(ascii_letters + digits + "_")),
+    "postalsafe": ascii_letters + digits + " .-#/",
+    "urlsafe": ascii_letters + digits + "-._~",
+    "domainsafe": ascii_letters + digits + "-",
+}
+
+_categories = {
+    "category_digit": _alphabets["digits"],
+    "category_not_digit": _alphabets["nondigits"],
+    "category_space": _alphabets["whitespace"],
+    "category_not_space": _alphabets["nonwhitespace"],
+    "category_word": _alphabets["word"],
+    "category_not_word": _alphabets["nonword"],
+}
+
+
+class RegexFactory:
+    def __init__(self, limit: int = 10, seed: Optional[int] = None) -> None:
+        self._limit = limit
+        self._cache: Dict[str, Any] = {}
+        self._random = Random(x=seed)
+
+        self._cases = {
```

Comment:
There are major additions to regexes in Python 3.11

Atomic grouping ((?>...)) and possessive quantifiers (*+, ++, ?+, {m,n}+) are now supported in regular expressions.

And it seems that this code doesn't support it. However, I think this can be addressed in a followup issue.

Reply:
a PR from you would be very welcome here, since i am really not a regex man.